### PR TITLE
[risk=low][RW-7634] Sync user access tiers when admin updates affiliation

### PR DESCRIPTION
**Pull request by jmthibault79**
Switch back and forth betw an inst which provides CT and one which does not (as admin)

Admin_Affiliation

Now switch contact emails within an inst, where one email grants RT+CT access and the other only RT

Admin_Contact_Email

Does not complete 7634 b/c a sync is not triggered when we save the institution itself.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/dao/UserService.java [hunk 1]
```diff
@@ -40,8 +40,6 @@ DbUser createUser(
       DbDemographicSurvey dbDemographicSurvey,
       DbVerifiedInstitutionalAffiliation dbVerifiedAffiliation);
 
-  DbUser updateUserWithConflictHandling(DbUser user);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserService.java:43]: no longer used

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 2]
```diff
@@ -420,24 +420,6 @@ public DbUser createUser(
...
 
-  /**
-   * Save updated dbUser object and transform ObjectOptimisticLockingFailureException into
-   * ConflictException
-   *
-   * @param dbUser
-   * @return
-   */
-  @Override
-  public DbUser updateUserWithConflictHandling(DbUser dbUser) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:431]: `updateUserWithRetries()` does this plus a lot more, so this is no longer needed.

File: api/src/main/java/org/pmiops/workbench/profile/ProfileService.java [hunk 3]
```diff
@@ -251,31 +251,34 @@ public DbUser updateProfile(
...
+    newAffiliation.setUser(dbUser);
 
-    // This is 1:1 in terms of our current usage, but not modelled that way (aside from the unique
-    // constraint on user_id).
-
-    // If this user already has an affiliation, replace the ID of hte
+    // If this user already has an affiliation, set the ID of the newAffiliation to it
     verifiedInstitutionalAffiliationDao
-        .findFirstByUser(user)
+        .findFirstByUser(dbUser)
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/profile/ProfileService.java:276]: so updatedProfile.getVerifiedInstitutionalAffiliation() doesn't have the id already?

Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/profile/ProfileService.java:276]: nvm its an existing code

**Review by NehaBroad — APPROVED** [review]

**Pull request merged**